nbextensions/bulkImportCell/paramsWidget.js

```diff
@@ -0,0 +1,551 @@
+            ]);
+        
+            const content = form({ dataElement: 'input-widget-form' }, formContent);
+            return {
+                content: content,
+                events: events
+            };
+        }
+
+        // MESSAGE HANDLERS
+        function doAttach(node) {
+            container = node;
+            ui = UI.make({
+                node: container,
+                bus: bus
+            });
+            let layout = renderLayout();
+            container.innerHTML = layout.content;
+            layout.events.attachEvents(container);
+
+            places = {
+                parameterFields: ui.getElement('parameter-fields'),
+                advancedParameterFields: ui.getElement('advanced-parameter-fields')
+            };
+        }
+
+        // EVENTS
+        function attachEvents() {
+            bus.on('reset-to-defaults', function () {
+                widgets.forEach(function (widget) {
+                    widget.bus.emit('reset-to-defaults');
+                });
+            });
+
+            bus.on('toggle-advanced', function () {
+                settings.showAdvanced = !settings.showAdvanced;
+                renderAdvanced();
+            });
+
+            runtime.bus().on('workspace-changed', function () {
+                // tell each input widget about this amazing event!
+                widgets.forEach(function (widget) {
+                    widget.bus.emit('workspace-changed');
+                });
+            });
+        }
+
+        function makeParamsLayout(params) {
+            let view = {},
+                paramMap = {};
+
+            const orderedParams = params.map(function (param) {
+                paramMap[param.id] = param;
+                return param.id;
+            });
+
+            const layout = orderedParams.map(function (parameterId) {
+                let id = html.genId();
+                view[parameterId] = {
+                    id: id
+                };
+
+                return div({
+                    id: id,
+                    dataParameter: parameterId
+                });
+            }).join('\n');
+
+            return {
+                content: layout,
+                layout: orderedParams,
+                params: params,
+                view: view,
+                paramMap: paramMap
+            };
+        }
+
+        // LIFECYCLE API
+        function renderParameters() {
+            // First get the app specs, which is stashed in the model,
+            // with the parameters returned.
+            // Separate out the params into the primary groups.
+            const appSpec = model.getItem('appSpec');
+
+            return Promise.try(function () {
+                const params = model.getItem('parameters');
+                let parameterParams = makeParamsLayout(
+                    params.layout.filter(function (id) {                    
+                        const original = params.specs[id].original;
```

Comment:
this filter block is where we check whether something is a parameter field, file input, or output. for files we look at `dynamic_dropdown_options.data_source` and for output it's `text_options.is_output_name` open to any input here on a better way to do this, especially if there are different app spec implementations i need to be aware of